engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -682,6 +690,26 @@ abstract class SemanticRole {
     }
   }
 
+  void _updateControls() {
+    if (semanticsObject.hasControlsNodes) {
+      semanticsObject.owner.addOneTimePostUpdateCallback(() {
+        final List<String> elementIds = <String>[];
```

Comment:
Here and elsewhere, feel free to omit local variable types, such as `final elementIds = <String>[]` is just fine, especially in new code.